tools/sil-opt/SILOpt.cpp

```diff
@@ -318,7 +330,12 @@ int main(int argc, char **argv) {
   } else if (OptimizationGroup == OptGroup::Performance) {
     runSILOptimizationPasses(*CI.getSILModule());
   } else {
-    runCommandLineSelectedPasses(CI.getSILModule());
+    auto *SILMod = CI.getSILModule();
+    {
+      auto T = irgen::createIRGenModule(SILMod, getGlobalLLVMContext());
+      runCommandLineSelectedPasses(SILMod, T.second);
+      irgen::deleteIRGenModule(T);
+    }
```

Comment:
Question. I am worried about being able to mix IRGen and Performance passes. What is to stop me from doing so and if I do so what will happen? Should we have an additional SIL stage or something like that? (Just thinking out loud).

One could even argue that perhaps we need a separate entry point (like autolink is to the driver that only can optimize lowered SIL) or something like that.

Like I said I am just thinking out loud (unsure).